pkg/controller/etcdproxy/etcdproxy_controller.go

```diff
@@ -42,74 +46,96 @@ import (
 	samplescheme "github.com/xmudrii/etcdproxy-controller/pkg/client/clientset/versioned/scheme"
 	informers "github.com/xmudrii/etcdproxy-controller/pkg/client/informers/externalversions/etcd/v1alpha1"
 	listers "github.com/xmudrii/etcdproxy-controller/pkg/client/listers/etcd/v1alpha1"
+	"k8s.io/apimachinery/pkg/util/intstr"
 )
 
-const controllerAgentName = "etcdstorage-controller"
+const controllerAgentName = "etcdproxy-controller"
 
 const (
 	// SuccessSynced is used as part of the Event 'reason' when a EtcdStorage is synced
 	SuccessSynced = "Synced"
 	// ErrResourceExists is used as part of the Event 'reason' when a EtcdStorage fails
-	// to sync due to a Deployment of the same name already existing.
+	// to sync due to a ReplicaSet of the same name already existing.
 	ErrResourceExists = "ErrResourceExists"
+	// ErrUnknown is used as part of the Event 'reason' when a EtcdStorage fails
+	// to get, create, or update resource.
+	ErrUnknown = "ErrUnknown"
 
 	// MessageResourceExists is the message used for Events when a resource
-	// fails to sync due to a Deployment already existing
+	// fails to sync due to a ReplicaSet already existing
 	MessageResourceExists = "Resource %q already exists and is not managed by EtcdStorage"
 	// MessageResourceSynced is the message used for an Event fired when a EtcdStorage
 	// is synced successfully
 	MessageResourceSynced = "EtcdStorage synced successfully"
 )
 
-// Controller is the controller implementation for EtcdStorage resources
-type Controller struct {
+// EtcdConnectionInfo type is used to wire information
+// used by controller to create ReplicaSets.
+type EtcdConnectionInfo struct {
```

Comment:
all this is about the core etcd? Better use core in the variable names.

Reply:
What do you think about going with `EtcdProxyControllerOptions`?

Reply:
we have the pattern elsewhere (apiserver, controller manager) that *Options is used for structs that are linked to flags. Compare cmd/kube-controller-manager. Are these all options for the controller or just the info for the etcdproxy pods lateron?

Reply:
For now only for etcdproxy pods. Maybe `EtcdProxyOptions` without `Controller`? Or something not related to `Options`?

Reply:
if it is just for the etcdproxy, EtcdProxyOptions sounds good. Alternatively, also CoreEtcdOptions would be good if it is only about the core etcd connection credentials.

Reply:
Sounds good to me. As it's only about the core etcd for now, I went with `CoreEtcdOptions`.